app/models/payment_provider_setup.rb

```diff
@@ -1,16 +1,35 @@
 require "flex_commerce_api/api_base"
+
+# @module FlexCommerce
 module FlexCommerce
   #
+  # @module PaymentProviderSetup
   # A flex commerce PaymentProvider model
   #
-  # This model provides access to the flex commerce PaymentProvider which is used to
-  # define a payment gateway and provide instructions to the front end application
-  # on how to set up the gateway in a browser (i.e. redirect to paypal etc..)
-  #
+  # This model is used to define a payment gateway and provide instructions
+  # to the front end application on how to set up the gateway in a browser 
+  # (i.e. redirect to paypal etc..)
   #
   #
-  class PaymentProviderSetup < FlexCommerceApi::ApiBase
-    belongs_to :payment_provider
+  module PaymentProviderSetup
 
+    class << self; attr_accessor :setup_type, :redirect_url; end
+    
+    # @method call
+    # 
+    # Triggers call to PaypalExpress::Setup service class to setup paypal
+    # 
+    # @return [FlexCommerce::PaymentAdditionalInfo]
+    def self.call(cart:, ip_address:, success_url:, cancel_url:, allow_shipping_change:, callback_url:, use_mobile_payments:)
+      # Call PaypalExpress::Setup service class, which will talk to active merchant gem for setup process
+      setup ||= ::FlexCommerce::Payments::PaypalExpress::Setup.new(cart: cart,
```

Comment:
`setup ||=` doesn't do anything here, remove it.

Reply:
Infact, we should drop this module altogether and simply have the front-end call `::FlexCommerce::Payments::PaypalExpress::Setup` directly.